### PR TITLE
update the style of the map viewer layout MA-385 

### PR DESCRIPTION
**Pull request by mihai-sysbio**
In this PR the aim is to refactor/drastically improve the layout of the Map Viewer on both mobile and desktop views. Any remaining bugs, eg the 2D map not showing up on mobile, should be reported in MA-897 and resolved separately.

### CODE REVIEW

File: frontend/src/components/explorer/MapViewer.vue [hunk 1]
```diff
@@ -223,15 +231,15 @@ export default {
...
-    padding: 10px;
-    border-radius: 5px;
-    background: rgba(22, 22, 22, 0.8);
-  }
+      position: absolute;
+      z-index: 10;
+      padding: 10px;
+      border-radius: 5px;
+      background: rgba(22, 22, 22, 0.8);
+    }
```
Comment (e0) [inline frontend/src/components/explorer/MapViewer.vue:242]: These indents seem like they are not needed.

File: frontend/src/components/explorer/MapViewer.vue [hunk 2]
```diff
@@ -243,13 +251,46 @@ export default {
       }
     }
   }
+
+  @media screen and (max-width: $tablet) {
+    &.ordered-mobile {
+      display: flex;
+      flex-flow: column;
+    }
+    .om-1 {
```
Comment (e0) [inline frontend/src/components/explorer/MapViewer.vue:260]: Just curious, what does the `om` stand for?

File: frontend/src/style/vars.scss [hunk 3]
```diff
@@ -25,6 +25,6 @@ $grey: hsl(0, 0%, 48%);
 $grey-lighter: hsl(0, 0%, 86%);
 $black: hsl(0, 0%, 4%);
 
-// Custom variables for fullheight Map Viewer
-$navbar-height: 2.5rem;
+// Custom variables for fullheight Map Viewer, same as Bulma defaults
```
Comment (e0) [inline frontend/src/style/vars.scss:28]: Can we these lines if they are the same as Bulma defaults?

**Review by e0 — APPROVED** [review]
Have added some notes in https://chalmersysbiocsb.atlassian.net/browse/MA-897.

File: frontend/src/components/explorer/MapViewer.vue [hunk 4]
```diff
@@ -161,6 +168,9 @@ export default {
     this.loadMapFromParams();
   },
   methods: {
+    dimensionalState(showing2D) {
+      return showing2D ? '2d' : '3d';
+    },
```
Comment (e0) [inline frontend/src/components/explorer/MapViewer.vue:173]: This is a good refactor and I think we could even go one step further and have it in a Vuex getter. Then we wouldn't need to pass it as a param. Not necessary now though.

File: frontend/src/components/explorer/MapViewer.vue [hunk 5]
```diff
@@ -11,24 +11,31 @@
...
+          <div id="dimensionToggle" class="buttons has-addons is-centered padding-mobile"
+               :title="`Switch to ${dimensionalState(!showing2D) }`"
+               @click="$store.dispatch('maps/toggleShowing2D')" tooltip="haha">
+            <button v-for="dim in [true, false]" :key="dim"
+                    class="button"
+                    :class="dim === showing2D ? 'is-selected is-primary has-text-weight-bold' : 'is-light'">
+              <span v-if="dim === showing2D" class="icon">
+                <i class="fa fa-check-square-o"></i>
+              </span>
+              <span v-if="dim !== showing2D">Switch to&nbsp;</span>
```
Comment (e0) [inline frontend/src/components/explorer/MapViewer.vue:23]: This sometimes causes buttons to be in two rows but I think it's negligible for now.

Comment (mihai-sysbio) [reply frontend/src/components/explorer/MapViewer.vue:23]: I think it's the `min-width` causing that, but removing that makes the text go outside the borders of the buttons; indeed, negligible for now.

Comment (e0) [reply frontend/src/components/explorer/MapViewer.vue:23]: Yes I meant specifically the change in text. It might fit without the “Switch to” addition.

**Comment by mihai-sysbio** [comment]
I have to admit, I really like this new map menu, and I'm happy to have the oldest unresolved issue behind us.

File: frontend/src/components/explorer/MapViewer.vue (hunk 5 above)
Comment (mihai-sysbio) [reply frontend/src/components/explorer/MapViewer.vue:23]: I tried, but it had a `width` that had to be overridden, so it made no difference. I kept that size for `min-width` instead.

**Pull request merged**